### PR TITLE
docs(spin/install): updates per v1.1.0

### PR DESCRIPTION
**Pull request by vdice**
- Updates the install steps per the next anticipated v1.1.0 release

Creating as a reminder; will keep in draft and add the appropriate label to be sure it isn't merged beforehand.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [x] Has run `npm run build-index`
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)

### CODE REVIEW

File: content/spin/install.md [hunk 1]
```diff
@@ -136,14 +136,15 @@ If you have [`cargo`](https://doc.rust-lang.org/cargo/getting-started/installati
 
 <!-- @selectiveCpy -->
 
-<pre class="bash spin-install" id="spin-install-source-cargo"><code>$ git clone https://github.com/fermyon/spin -b v0.10.0
+<pre class="bash spin-install" id="spin-install-source-cargo"><code>$ git clone https://github.com/fermyon/spin -b v1.1.0
```
Comment (dicej) [inline content/spin/install.md:139]: This won't work until we actually create a `v1.1.0` tag, of course.

**Review by dicej — APPROVED** [review]
Looks good, as long as we don't deploy it prior to the v1.1.0 release.

**Comment by karthik2804** [comment]
Should we wait on this until v1.1? I'm assuming most people who build from source probably build from main. Also i believe these changes would not break even if they build from an older commit.  Thoughts?

**Review by tpmccallum — APPROVED** [review]

**Comment by tpmccallum** [comment]
Thanks @vdice this is spot on for the new release.

File: content/spin/install.md [hunk 2]
```diff
@@ -136,14 +136,15 @@ If you have [`cargo`](https://doc.rust-lang.org/cargo/getting-started/installati
...
 $ cd spin
 $ rustup target add wasm32-wasi
+$ rustup target add wasm32-unknown-unknown
 $ cargo install --locked --path .
 $ spin --help
 </code></pre>
 
-> Please note: Spin v0.10.0 requires `wasmtime v5.0.0` which requires rustc 1.66.0 or newer. You can update Rust using the following command:
+> Please note: Spin v1.1.0 requires `wasmtime v7.0.0` which requires rustc 1.68.0 or newer. You can update Rust using the following command:
```
Comment (itowlson) [inline content/spin/install.md:147]: Consider dropping the "reason why" sentence. No user cares which dependency induces the version requirement _grin_. I also suggest qualifying "Spin requires" lest a reader's eyes slide past the section header and they get the impression that the Spin binary depends on having Rust installed.
 
```suggestion
> Please note: Installing Spin v1.1.0 from source requires Rust 1.68.0 or newer. You can update Rust using the following command:
```

**Comment by vdice** [comment]
(Moving into review as we gear up for the Spin 1.1.0 release)

**Pull request merged**